ampd/src/stellar/http_client.rs

```diff
@@ -15,12 +17,41 @@ pub enum Error {
     Client,
     #[error("invalid tx hash")]
     TxHash,
+    #[error("failed to deserialize tx id")]
+    FailedTxIdDeserialization,
+}
+
+#[derive(Debug, Clone, Eq, Hash, Deserialize, Serialize, PartialEq)]
+pub struct StellarTxId(#[serde(deserialize_with = "deserialize_stellar_tx_id")] String);
+
+/// Deserializes a Stellar tx id from a 0x-prefixed tx id.
+fn deserialize_stellar_tx_id<'de, D>(deserializer: D) -> Result<String, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let tx_id: String = String::deserialize(deserializer)?;
+    tx_id
+        .strip_prefix("0x")
+        .map(String::from)
+        .ok_or(report!(D::Error::custom(Error::FailedTxIdDeserialization)))
+}
+
+impl StellarTxId {
+    pub fn to_message_id(&self, event_index: u32) -> String {
+        format!("0x{}-{}", self.0.clone(), event_index)
```

Comment:
Is this the main change, we append 0x?

Reply:
nvm, I see what this is for. I don't think we should hardcode the format here like this, just use the existing message id format struct